### ISSUE TITLE
via-API templates - Principal Account details are no longer presented in the parameters.json 

### ISSUE THREAD

**Issue by vvt137** [F5Networks/f5-azure-arm-templates#163]
##  Description
Principal Account details are no longer presented in the parameters.json file for the following template (possibly, in other via-api templates):
https://github.com/F5Networks/f5-azure-arm-templates/tree/master/supported/failover/same-net/via-api/n-nic/existing-stack/byol

During deployment the template seems to default to using the user's Azure account as a Principle, which causes issues.

The previous versions of the template's paratemeter.json file contained the in parameters in question (at the very bottom of the file):
        "tenantId": {
            "value": ""
        },
        "clientId": {
            "value": ""
        },
        "servicePrincipalSecret": {
            "value": ""
        }

## Template
https://github.com/F5Networks/f5-azure-arm-templates/tree/master/supported/failover/same-net/via-api/n-nic/existing-stack/byol

## Severity Level
2

**Comment by JeffGiroux** [comment]
Agree, this is confusing. The new version states the following new line...
This solution requires roleAssignment creation and therefore, end-user must have sufficient permissions to create roleAssignments

The README also has the existing line...
This template requires a service principal. See the Service Principal Setup section for details, including required permissions.

There is no place to enter service principal information nor specify a role.

**Comment by JeffGiroux** [comment]
To note, this template deploy successfully and failover works successfully. All cloud objects moved when doing failover. The VMs get created and the role assigned is "Contributor". Docs can be more clear regarding service principal as README still states it is required.

**Comment by vvt137** [comment]
I suspect the situation I have is a rather typical one... My Azure account has "Contributor" privileges, but it does NOT have rights to create other accounts, so the deployment fails with error messages "...The client '<name>' with object id '<id>' does not have permission to perform action 
'Microsoft.Authorization/roleAssignments/write' at scope...". In the previous version of the template it was not a problem as I asked an admin to assign a principal/service account I created the necessary privileges and then I just populated the name/secret of that account into the template's variables.

**Comment by vvt137** [comment]
Is this change/limitation caused by the way "Cloud Failover Extension" works? Or is this related to way template itself is structured? If later, would it be technically possible to reintroduce the ability to use a pre-created account that version 7.3.0 (and prior versions) had?

**Comment by JeffGiroux** [comment]
Cloud Failover extension makes use of managed identities (roles) assigned to VMs. This avoids the need for SP accounts and credentials laying around in code.

**Comment by vvt137** [comment]
Yes, I can see that this is a better approach from security perspective. Still, it could be highly problematic for a person deploying the F5 templates to have "assign role to managed identity" privileges. Would it be possible to introduce a "postponed role assignment" trigger in the template's parameters? If this parameter is set to False (default) the template would be executed as-is, if it is set to True the template would not attempt to modify the roles assigned to the F5 VM identities, so that the person implementing the template can approach the security/admin team and ask them to assign the necessary roles to these (already created) VM identities/accounts manually.

**Comment by shyawnkarim** [comment]
Thanks for reaching out to us with this issue. I've created ESECLDTPLT-1961 to add this parameter back into our templates.

**Comment by Charding2008** [comment]
Experiencing this same issue with Service principal  the deployment fails with error messages "...The client '' with object id '' does not have permission to perform action
'Microsoft.Authorization/roleAssignments/write' at scope...".
This worked fine previously but not yet ready to move to Cloud Formation template.

**Comment by Charding2008** [comment]
> Thanks for reaching out to us with this issue. I've created ESECLDTPLT-1961 to add this parameter back into our templates.

any update on this?

**Comment by shyawnkarim** [comment]
@Charding2008 This issue is still in our backlog and has not been scheduled yet.

**Comment by grf5** [comment]
I have a project with the same limitations. The Azure admins created a role with the proper rights, but I cannot figure out how to simply use an existing role in the template versus having to create one.
